src/edu/csus/ecs/pc2/services/core/JSONUtilities.java

```diff
@@ -337,4 +341,31 @@ public static String getLanguageIndexString(IInternalContest contest, ElementId
         return Integer.toString(getLanguageIndex(contest, elementId));
     }
 
+    /**
+     * Returns a Map containing the key/value elements in the specified JSON string.
+     * This method uses the Jackson {@link ObjectMapper} to perform the conversion from the JSON
+     * string to a Map.  Note that the ObjectMapper recurses for nested JSON elements, returning
+     * a appropriate Object in the Map under the corresponding key string.
+     * 
+     * @param jsonString a JSON string to be converted to a Map
+     * @return a Map mapping the keys in the JSON string to corresponding values, or null if the input
+     *          String is null or if an exception occurs while converting the JSON to a Map.
+     */
+    @SuppressWarnings("unchecked")
+    public static Map<String, Object> getMap(String jsonString) {
```

Comment:
Probably should update RemoteEventFeedMonitor to use this method since it has the same method.

Reply:
Noted, out of scope though.

Reply:
I agree, this is something that should be done but is out of scope for this particular issue.   Hopefully, one of us will at some point remember to file a separate Issue for this...

Reply:
There is a bunch of refactoring to be done.   Some are marked with // TODO REFACTOR